python/fledge/services/core/api/service.py

```diff
@@ -183,29 +185,72 @@ async def add_service(request):
+                    ['name', '=', name]).payload()
+                result = await storage.query_tbl_with_payload('packages', select_payload)
+                response = result['rows']
+                if response:
+                    exit_code = response[0]['status']
+                    if exit_code == -1:
+                        msg = "{} package installation already in progress".format(name)
+                        return web.HTTPTooManyRequests(reason=msg, body=json.dumps({"message": msg}))
+                    # Remove old entry from table for other cases
+                    delete_payload = PayloadBuilder().WHERE(['action', '=', action]).AND_WHERE(
+                        ['name', '=', name]).payload()
+                    await storage.delete_from_tbl("packages", delete_payload)
+
+                # Check If requested service is already installed and then return immediately
+                services = get_service_installed()
+                svc_name = name.split('fledge-')[1].split('-')[1]
+                for s in services:
+                    if s == svc_name:
+                        msg = "{} package is already installed".format(name)
+                        return web.HTTPNotModified(reason=msg, body=json.dumps({"message": msg}))
+
+                # Check If requested service is available for configured repository
+                services, log_path = await common.fetch_available_packages("service")
+                if name not in services:
+                    raise KeyError('{} service is not available for the given repository'.format(name))
+                # Insert record into Packages table
+                insert_payload = PayloadBuilder().INSERT(id=str(uuid.uuid4()), name=name, action=action, status=-1,
+                                                         log_file_uri="").payload()
+                result = await storage.insert_into_tbl("packages", insert_payload)
+                response = result['response']
+                if response:
+                    # GET id from Packages table to track the installation response
+                    select_payload = PayloadBuilder().SELECT("id").WHERE(['action', '=', action]).AND_WHERE(
+                        ['name', '=', name]).payload()
+                    result = await storage.query_tbl_with_payload('packages', select_payload)
+                    response = result['rows']
+                    if response:
+                        pn = "{}-{}".format(action, name)
+                        uid = response[0]['id']
+                        p = multiprocessing.Process(name=pn, target=install.install_package_from_repo,
+                                                    args=(name, pkg_mgt, version, uid, storage))
+                        p.daemon = True
+                        p.start()
+                        _logger.info("{} service started...".format(name))
```

Comment:
what started :)

Reply:
installation text needs to be placed before

Reply:
Yeah, let's log the `msg` itself (the next line)